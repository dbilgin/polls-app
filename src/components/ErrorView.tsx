import React from 'react';
import {StyleSheet, View, Text} from 'react-native';

const ErrorView: React.FC = () => {
  return (
    <View style={styles.container}>
      <Text>An error has occurred.</Text>
    </View>
  );
};

export default React.memo(ErrorView);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
