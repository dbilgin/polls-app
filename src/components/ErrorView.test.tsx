import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ErrorView from './ErrorView';

Enzyme.configure({adapter: new Adapter()});

describe('ErrorView', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<ErrorView />);
      expect(component).toMatchSnapshot();
    });
  });
});
