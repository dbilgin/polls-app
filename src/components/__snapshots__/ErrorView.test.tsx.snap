// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    An error has occurred.
  </Text>
</View>
`;
